Description: Improve cross building: Use packages mir_wayland_generator during cross builds.
Author: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>
Origin: vendor
Forwarded: no
Bug-Debian: https://bugs.debian.org/1069578
Last-Update: 2025-02-20

--- a/cmake/MirCommon.cmake
+++ b/cmake/MirCommon.cmake
@@ -310,6 +310,14 @@ function (mir_check_no_unreleased_symbol
   add_dependencies(${DEPENDENT_TARGET} ${TARGET_NAME})
 endfunction()
 
+set(
+  MIR_WAYLAND_GENERATOR_EXECUTABLE
+  "${CMAKE_BINARY_DIR}/bin/mir_wayland_generator"
+  CACHE
+  STRING
+  "Location of an externally supplied mir_wayland_generator executable"
+)
+
 function (mir_generate_protocol_wrapper TARGET_NAME NAME_PREFIX PROTOCOL_FILE)
   if (NAME_PREFIX STREQUAL "")
     set(NAME_PREFIX "@") # won't match anything
@@ -322,9 +330,9 @@ function (mir_generate_protocol_wrapper
     OUTPUT "${OUTPUT_PATH_HEADER}" "${OUTPUT_PATH_SRC}"
     VERBATIM
     COMMAND "sh" "-c"
-    "${CMAKE_BINARY_DIR}/bin/mir_wayland_generator ${NAME_PREFIX} ${PROTOCOL_PATH} header > ${OUTPUT_PATH_HEADER}"
+    "${MIR_WAYLAND_GENERATOR_EXECUTABLE} ${NAME_PREFIX} ${PROTOCOL_PATH} header > ${OUTPUT_PATH_HEADER}"
     COMMAND "sh" "-c"
-    "${CMAKE_BINARY_DIR}/bin/mir_wayland_generator ${NAME_PREFIX} ${PROTOCOL_PATH} source > ${OUTPUT_PATH_SRC}"
+    "${MIR_WAYLAND_GENERATOR_EXECUTABLE} ${NAME_PREFIX} ${PROTOCOL_PATH} source > ${OUTPUT_PATH_SRC}"
     DEPENDS mir_wayland_generator "${PROTOCOL_PATH}"
   )
   target_sources("${TARGET_NAME}" PRIVATE "${OUTPUT_PATH_HEADER}" "${OUTPUT_PATH_SRC}")
--- a/tests/acceptance-tests/wayland-generator/CMakeLists.txt
+++ b/tests/acceptance-tests/wayland-generator/CMakeLists.txt
@@ -5,9 +5,9 @@ add_custom_command(
   OUTPUT "${GENERATED_HEADER}" "${GENERATED_SRC}"
   VERBATIM
   COMMAND "sh" "-c"
-  "${CMAKE_BINARY_DIR}/bin/mir_wayland_generator wl_ ${PROTOCOL_PATH} header > ${GENERATED_HEADER}"
+  "${MIR_WAYLAND_GENERATOR_EXECUTABLE} wl_ ${PROTOCOL_PATH} header > ${GENERATED_HEADER}"
   COMMAND "sh" "-c"
-  "${CMAKE_BINARY_DIR}/bin/mir_wayland_generator wl_ ${PROTOCOL_PATH} source > ${GENERATED_SRC}"
+  "${MIR_WAYLAND_GENERATOR_EXECUTABLE} wl_ ${PROTOCOL_PATH} source > ${GENERATED_SRC}"
   DEPENDS mir_wayland_generator "${PROTOCOL_PATH}"
 )
 add_custom_target(wayland_generator_test_generated_files ALL DEPENDS "${GENERATED_SRC}" "${GENERATED_HEADER}")
